lib/resolveTaskFn.js

```diff
@@ -56,18 +56,22 @@ const handleOutput = (command, result, ctx, isError = false) => {
  * checks the context.
  */
 const interruptExecutionOnError = (ctx, execaChildProcess) => {
+  let loopIntervalId
+
   async function loop() {
     if (ctx.errors.size > 0) {
+      clearInterval(loopIntervalId)
+
       const ids = await pidTree(execaChildProcess.pid)
```

Comment:
If the execa child process is already dead at this point, `pidTree` will throw an error because the pid won't be found. Should we just do something like:

```suggestion
      const ids = await pidTree(execaChildProcess.pid).catch(() => [])
```

Reply:
Naturally that was also my first attempt to fix it. Unfortunately it all boils down to much more hussle around catching stuff.

First we should catch `pidTree` (silently, or we can check specifically the thrown `new Error('No matching pid found')`).

Then we come to `SIGTERM` broadcasting. `process.kill` will also throw an error if signal recipient do not exist. Again, we can catch it silently, or check for `error.code === 'ESRCH'` ([ESRCH](https://man7.org/linux/man-pages/man3/errno.3.html) - "No such process")

It's just a lot of convoluted code to handle _the situation_ after the fact. A code that will have be read and maintained at some point.
Instead we'd better prevent occurrence of _the situation_ in the first place. So after a few iterations (mentioned in the oroginal PR comment) I came up with what we have here now.

Here's some comparison on how much literally "cleaner" and straightforward the event loop becomes if we terminate the "interval watcher" on the first error occurrence. 
<span><img src="https://user-images.githubusercontent.com/56614219/163631569-34587922-6006-4ac9-af75-9d4a3036b698.png" width="45%" alt="event loop before the fix" /></span> <span><img src="https://user-images.githubusercontent.com/56614219/163631574-abdcd222-27ff-44d8-8685-f3f1b2585427.png" width="45%" alt="event loop after the fix" /></span>

There's huge amount of unnecessary error handling during the clean-up. Each "🚫fail" have to be catched and processed. As I've mentioned it originally: _just messy overall_.
#
Now if you want we can also review some risks of that solution. Such scenarios, for which preemptive `clearTimeout` would cause a failure, while "kill-kill-catch-kill-catch" approach would succeed.
The only case I can imagine, is when we create A LOT of processes, but execution is interrupted even before all of them has born (but they continue launching). Or it was just interrupted immediately, even before the current event loop task finished. Some veery edgy cases.

I am not aware of a [a]synchronousness degree of process launching with `execa` per se, but nevertheless I do not expect that such occasion is even possible in a real world scenario.

Reply:
I have made some improvements related to this in https://github.com/okonet/lint-staged/pull/1173

Reply:
More specifically:

- added try/catch to `pid-tree` and `process.kill`
- replaced `setInterval` with Node.js `EventEmitter`
- waiting for the task kill promise in the main handler before proceeding (mostly theoretic, but doesn't hurt)